File: Shared/Preferences Controllers/PianoPreferencesController.m
//
//  PianoPreferencesController.m
//  PPMacho
//
//  Created by C.W. Betts on 7/26/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "PianoPreferencesController.h"
#import "UserDefaultKeys.h"


@implementation PianoPreferencesController

+ (instancetype)newPreferenceView
{
	return [[self alloc] init];
}

- (NSString*)identifier
{
	return PPPianoPrefID;
}

- (NSString*)toolbarItemLabel
{
	return NSLocalizedStringFromTable(@"Piano", @"PreferenceNames", @"Piano");
}

- (NSImage*)toolbarItemImage
{
	return nil;
}

- (instancetype)init
{
	if (self = [super initWithNibName:@"PianoPrefs" bundle:nil]) {
		[self setTitle:NSLocalizedStringFromTable(@"Piano", @"PreferenceNames", @"Piano")];
	}
	return self;
}

- (void)awakeFromNib
{
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
	
	defaults = nil;
}

- (BOOL)hasResizableWidth
{
	return NO;
}

- (BOOL)hasResizableHeight
{
	return NO;
}

@end

File: Shared/Preferences Controllers/PianoPreferencesController.swift
//
//  PianoPreferencesController.swift
//  PPMacho
//
//  Created by C.W. Betts on 8/12/14.
//
//

import Cocoa

class PianoPreferencesController: NSViewController, PlayerPROPreference {
	required init(coder: NSCoder!) {
		super.init(coder: coder)
	}
	
	required override init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: NSBundle!) {
		super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
	}
	
	class func newPreferenceView() -> Self {
		var nullBundle: NSBundle! = nil
		var ourself = self(nibName: "PianoPrefs", bundle: nullBundle)
		
		var aTitle = NSLocalizedString("Piano", tableName:"PreferenceNames", comment: "Piano");
		ourself.title = aTitle
		return ourself
	}

    override func awakeFromNib() {
        super.awakeFromNib()
        // Do view setup here.
    }
    
}
